dnf/cli/cli.py

```diff
@@ -289,7 +291,40 @@ def gpgsigcheck(self, pkgs):
                 logger.critical(msg)
             raise dnf.exceptions.Error(_("GPG check FAILED"))
 
-    def check_updates(self, patterns=(), reponame=None, print_=True):
+    def latest_changelogs(self, package):
+        """Return list of changelogs for package newer then installed version"""
+        newest = None
+        # find the date of the newest changelog for installed version of package
+        # stored in rpmdb
+        for mi in self._rpmconn.readonly_ts.dbMatch('name', package.name):
+            changelogtimes = mi[rpm.RPMTAG_CHANGELOGTIME]
+            if changelogtimes:
+                newest = datetime.date.fromtimestamp(changelogtimes[0])
+                break
+        chlogs = [chlog for chlog in package.changelogs
+                  if newest is None or chlog['timestamp'] > newest]
+        return chlogs
+
+    def format_changelog(self, changelog):
+        """Return changelog formated as in spec file"""
+        chlog_str = '* %s %s\n%s\n' % (changelog['timestamp'].strftime("%a %b %d %Y"),
```

Comment:
What if full timestamps are being used in the RPM package changelogs? That's been supported since RPM 4.14.0, and there are definitely users of that feature.